src/routes/photos.router.ts

```diff
@@ -0,0 +1,58 @@
+import * as express from "express";
+
+const catsList = [
+  "https://founded.media/hiring/photos/cats/14157413946_fea785b4d6_k.jpg",
+  "https://founded.media/hiring/photos/cats/16175483119_bd7374d8a8_h.jpg",
+  "https://founded.media/hiring/photos/cats/13901304865_a444cf4d34_k.jpg",
+  "https://founded.media/hiring/photos/cats/8311701653_49ed80202c_k.jpg",
+  "https://founded.media/hiring/photos/cats/13336301695_3c06dd41cc_k.jpg",
+  "https://founded.media/hiring/photos/cats/38679744435_66279af67c_k.jpg",
+  "https://founded.media/hiring/photos/cats/6393395037_9cda69da1a_b.jpg",
+  "https://founded.media/hiring/photos/cats/6977309082_44102ddf51_b.jpg",
+  "https://founded.media/hiring/photos/cats/11477923503_bbdf86387d_b.jpg",
+  "https://founded.media/hiring/photos/cats/4481336172_7f464f180d_b.jpg",
+];
+
+const sharksList = [
+  "https://founded.media/hiring/photos/sharks/11261840124_dc9ac72bbe_b.jpg",
+  "https://founded.media/hiring/photos/sharks/513197047_2f861d56cb_b.jpg",
+  "https://founded.media/hiring/photos/sharks/2989909952_b59500107e_o.jpg",
+  "https://founded.media/hiring/photos/sharks/4107884442_3baf8985f2_b.jpg",
+  "https://founded.media/hiring/photos/sharks/3822452418_ffa66da89d_o.jpg",
+  "https://founded.media/hiring/photos/sharks/3800013954_20fea3a9c9_b.jpg",
+  "https://founded.media/hiring/photos/sharks/7109693941_250fc6b246_k.jpg",
+  "https://founded.media/hiring/photos/sharks/8592704407_75c3c7ff53_h.jpg",
+  "https://founded.media/hiring/photos/sharks/14730744390_cebc28aa86_k.jpg",
+  "https://founded.media/hiring/photos/sharks/4936728723_91da549b05_b.jpg",
+];
+
+const photosRouter = express.Router();
+
+const shuffle = (array: any[]) => {
+  return array.slice().sort(() => Math.random() - 0.5);
+};
+
+photosRouter.get("/", (req, res, next) => {
+  const result = [...catsList, ...sharksList];
+
+  res.status(200).send({
+    success: true,
+    data: shuffle(result),
+  });
+});
+
+photosRouter.get("/cats", (req, res, next) => {
+  res.status(200).send({
+    success: true,
+    data: catsList,
+  });
+});
+
+photosRouter.get("/sharks", (req, res, next) => {
+  res.status(200).send({
+    success: true,
+    data: sharksList,
+  });
+});
```

Comment:
How could you refactor this such that there is only 1 endpoint?

Reply:
This can be done with query parameters, especially if we would want to have more photo types in the future, which seems likely. 

We can add a filters query parameter array, specifying the shark filter only returns sharks, specifying shark and cat will return both, specifying nothing will just return everything. The request would look something like this: -

```/api/photos/?filters[]=shark&filters[]=cat```

In express, this can be read through the request parameter. With the above string, `req.query` in `photosRouter.get("/",...` would look like this: -

```javascript
{ filters: [ 'shark', 'cat' ] } 
```

And we can formulate our response accordingly. Happy to update PR to implement this!